code/main.py

```diff
@@ -351,7 +354,7 @@ def run_model(actor_critic, current_memory, envs,
     # Also, if action is discrete, convert it to one-hot vector
     current_memory['oneHotActions'] = utils.toOneHot(
         envs.action_space,
-        policy_return.action * masks.type(policy_return.action.type()))
+        policy_return.action * masks.type(policy_return.action.type()).to(device))
```

Comment:
Essentially the same question as above. I'm not sure how sacred manages setting devices.

Reply:
This is a good catch and some slopping coding on my part. The device is set in 2 places, once in random_variable.py and once in floorNav.yaml (the config variable called cuda). Inside main.py, the function general_config() sets the device variable according to what's in floorNav.yaml. I have to always make sure that the setting in random_variable and floorNav is the same. I think this is because the DVRL people didn't make sure their code was robust to using a different gpu than "auto".